go/vt/sqlparser/sql.y

```diff
@@ -6942,26 +6954,35 @@ set_list:
   }
 
 set_expression:
-  reserved_sql_id '=' ON
+  set_variable '=' ON
```

Comment:
Yeah, this is the way that the old code was doing it. I've been trying to find any valid `SET` queries that pass on mysql and fail on vitess because of this rule and I have not been able to find any. Since this is simpler, I'm going to leave it as is.